Source/WebCore/css/CSSGroupingRule.cpp

```diff
@@ -76,7 +76,17 @@ ExceptionOr<unsigned> CSSGroupingRule::insertRule(const String& ruleString, unsi
     }
 
     RefPtr styleSheet = parentStyleSheet();
-    RefPtr newRule = CSSParser::parseRule(ruleString, parserContext(), styleSheet ? &styleSheet->contents() : nullptr, CSSParser::AllowedRules::ImportRules, nestedContext());
+    auto nestedContextWithCurrentRule = [&] -> CSSParserEnum::NestedContext {
+        if (m_groupRule->isStyleRule()) {
+            ASSERT_NOT_REACHED(); // This is handled in CSSStyleRule.
+            return CSSParserEnum::NestedContextType::Style;
+        }
```

Comment:
Why not `ASSERT(!!m_groupRule->isStyleRule())`? Presumably the current behavior of calling `nestedContext()` is also fine?

Reply:
At some point, we should merge CSSStyleRule and CSSGroupingRule. The code is written like this to make this future merge easier hopefully.